File: backend/src/main/java/at/ac/tuwien/sepm/assignment/individual/dto/HorseDto.java
package at.ac.tuwien.sepm.assignment.individual.dto;

import at.ac.tuwien.sepm.assignment.individual.enums.Gender;

import java.time.LocalDate;

/**
 * Class for Horse DTOs
 * Contains all common properties
 */
public record HorseDto(Long id, String name, String description, LocalDate birthdate, Gender gender, Long ownerId, HorseDto damId, HorseDto sireId) {

}

File: frontend/src/app/dto/horse/horseDto.ts
import { Gender } from './gender';
import { OwnerDto} from '../owner/ownerDto';

export interface HorseDto {
  id?: number;
  name: string;
  description?: string;
  birthdate: Date;
  gender: Gender;
  owner?: OwnerDto;
  dam?: HorseDto;
  sire?: HorseDto;
}
